pyccel/codegen/printing/cwrappercode.py

```diff
@@ -0,0 +1,351 @@
+            cast_function = self.get_cast_function(used_names, 'complex_to_pycomplex', variable, collect_var)
+            return collect_var , cast_function
+
+        return variable, None
+
+    def _print_PyccelPyObject(self, expr):
+        return 'pyobject'
+
+    def _print_FuncCall(self, expr):
+        name = expr.name
+        args = ','.join(['{}'.format(self._print(a)) for a in expr.args])
+        return ('{name}({args})'.format(
+            name = name,
+            args = args))
+
+    def _print_PyArg_ParseTupleNode(self, expr):
+        name = 'PyArg_ParseTupleAndKeywords'
+        pyarg = expr.pyarg
+        pykwarg = expr.pykwarg
+        flags = expr.flags
+        args = ', '.join(['{}'.format(self._print(VariableAddress(a))) for a in expr.args])
+        if expr.args:
+            code = '{name}({pyarg}, {pykwarg}, "{flags}", {kwlist}, {args})'.format(
+                            name=name,
+                            pyarg=pyarg,
+                            pykwarg=pykwarg,
+                            flags = flags,
+                            kwlist = expr.arg_names.name,
+                            args = args)
+        else :
+            code ='{name}({pyarg}, {pykwarg}, "", {kwlist})'.format(
+                    name=name,
+                    pyarg=pyarg,
+                    pykwarg=pykwarg,
+                    kwlist = expr.arg_names.name)
+        return code
+
+    def _print_PyBuildValueNode(self, expr):
+        name = 'Py_BuildValue'
+        flags = expr.flags
+        args = ','.join(['{}'.format(self._print(VariableAddress(a)) for a in expr.args])
+        #to change for args rank 1 +
+        if expr.args:
+            code = '{name}("{flags}", {args})'.format(name=name, flags=flags, args=args)
+        else :
+            code = '{name}("")'.format(name=name)
+        return code
+
+    def _print_PyArgKeywords(self, expr):
+        arg_names = ', \n'.join(['"{}"'.format(a) for a in expr.arg_names] + [self._print(Nil())])
+        return ('static char *{name}[] = {{\n'
+                        '{arg_names}\n'
+                        '}};\n'.format(name=expr.name, arg_names = arg_names))
+
+    def _print_FunctionDef(self, expr):
+        # Save all used names
+        used_names = set([a.name for a in expr.arguments] + [r.name for r in expr.results] + [expr.name.name])
+
+        # Find a name for the wrapper function
+        wrapper_name = self.get_new_name(used_names.union(self._global_names), expr.name.name+"_wrapper")
+        self._function_wrapper_names[expr.name] = wrapper_name
+        self._global_names.add(wrapper_name)
+        used_names.add(wrapper_name)
+
+        # Collect local variables
+        wrapper_vars = [a for a in expr.arguments] + [r for r in expr.results]
+        python_func_args = Variable(dtype=PyccelPyObject(),
+                                 name=self.get_new_name(used_names, "args"),
+                                 is_pointer=True)
```

Comment:
We have this code a lot. I wonder if we should create a small helper function :
```
def get_new_PyObject(name, used_names):
    return Variable(dtype=PyccelPyObject(),
                                 name=self.get_new_name(used_names, name),
                                 is_pointer=True)
```
To reduce code duplication/make the code easier to read?

Reply:
yes that's a good idea